pkg/pod/script.go

```diff
@@ -211,3 +238,55 @@ cat > ${scriptfile} << '%s'
 func encodeScript(script string) string {
 	return base64.StdEncoding.EncodeToString([]byte(script))
 }
+
+func checkWindowsRequirement(steps []v1beta1.Step, sidecars []v1beta1.Sidecar) bool {
+	requiresWindows := false
+	// Detect windows shebangs
+	for _, step := range steps {
+		cleaned := strings.TrimSpace(step.Script)
+		if strings.HasPrefix(cleaned, "#!win") {
+			return true
+		}
+	}
+	// If no step needs windows, then check sidecars to be sure
+	if !requiresWindows {
```

Comment:
Given `requiresWindows` is never changed I don't think it's needed.  Simply running the loop over steps and then sidecars should be equivalent I think?